accessors/spanner/spannermetadataaccessor/clients/spanner_metadata_client.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package spannermetadataclient
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	sp "cloud.google.com/go/spanner"
+)
+
+var once sync.Once
+var spannermetadataClient *sp.Client
+
+var newClient = sp.NewClient
+
+func GetOrCreateClient(ctx context.Context, dbURI string) (*sp.Client, error) {
+	var err error
+	if spannermetadataClient == nil {
+		once.Do(func() {
+			spannermetadataClient, err = newClient(ctx, dbURI)
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create spanner database client: %v", err)
```

Comment:
`failed to create spanner metadata database client`

Reply:
done